doc/port-config-refactor/port-config-refactor-design.md

```diff
@@ -0,0 +1,36 @@
+# SONiC Port Configuration Refactor Design #
```

Comment:
I think the changing design part is OK, suggest adding a test plan section

Reply:
Added test plan part, please review